src/crimson/osd/shard_services.cc

```diff
@@ -59,7 +62,11 @@ seastar::future<> ShardServices::send_to_osd(
   } else {
     auto conn = cluster_msgr.connect(
         osdmap->get_cluster_addrs(peer).front(), CEPH_ENTITY_TYPE_OSD);
-    return conn->send(m);
+    if (conn) {
+      return conn->send(m);
+    } else {
+      throw std::runtime_error(fmt::format("error connecting to osd.{}", peer));
+    }
```

Comment:
Not related.

Reply:
Sorry, this is a left over, I'll correct it, sorry for this mistake....